diff --git a/CMakeLists.txt b/CMakeLists.txt
index 1d7d78a..de37c79 100644
--- a/CMakeLists.txt
+++ b/CMakeLists.txt
@@ -951,6 +951,7 @@ status("  Other third-party libraries:")
 if(WITH_IPP AND IPP_FOUND)
   status("    Use IPP:" "${IPP_LATEST_VERSION_STR} [${IPP_LATEST_VERSION_MAJOR}.${IPP_LATEST_VERSION_MINOR}.${IPP_LATEST_VERSION_BUILD}]")
   status("         at:" "${IPP_ROOT_DIR}")
+  set(CMAKE_INSTALL_RPATH "${CMAKE_INSTALL_RPATH}:${IPP_LIBRARY_DIRS}")
 else()
   status("    Use IPP:"   WITH_IPP AND NOT IPP_FOUND THEN "IPP not found" ELSE NO)
 endif()
diff --git a/cmake/OpenCVFindIPP.cmake b/cmake/OpenCVFindIPP.cmake
index db02e6a..abdd411 100644
--- a/cmake/OpenCVFindIPP.cmake
+++ b/cmake/OpenCVFindIPP.cmake
@@ -229,7 +229,7 @@ function(set_ipp_variables _LATEST_VERSION)
         endif()
 
         if (UNIX)
-            get_filename_component(INTEL_COMPILER_LIBRARY_DIR ${IPP_ROOT_DIR}/../lib REALPATH)
+            get_filename_component(INTEL_COMPILER_LIBRARY_DIR ${IPP_ROOT_DIR}/../compiler/lib REALPATH)
             if (IPP_X64)
                 if(NOT EXISTS ${INTEL_COMPILER_LIBRARY_DIR}/intel64)
                     message(SEND_ERROR "Intel compiler EM64T libraries not found")
